imap/imap.go

```diff
@@ -150,203 +63,176 @@ func Validate(s *models.IMAP) error {
+		return err
 	}
 
-	return client, nil
-}
+	defer imapClient.Logout()
 
-const dateFormat = "02-Jan-2006"
+	seqSet := new(imap.SeqSet)
+	seqSet.AddNum(seqs...)
 
-// findEmails will run a find the UIDs of any emails that match the search.:
-func findEmails(client Client, search string, since *time.Time) (*imap.Command, error) {
-	var specs []imap.Field
-	if len(search) > 0 {
-		specs = append(specs, search)
+	item := imap.FormatFlagsOp(imap.RemoveFlags, true)
+	err = imapClient.Store(seqSet, item, imap.SeenFlag, nil)
+	if err != nil {
+		return err
 	}
 
-	if since != nil {
-		sinceStr := since.Format(dateFormat)
-		specs = append(specs, "SINCE", sinceStr)
-	}
+	return nil
+
+}
 
-	// get headers and UID for UnSeen message in src inbox...
-	cmd, err := imap.Wait(client.UIDSearch(specs...))
+// DeleteEmails will delete emails from the supplied slice of SeqNums
+func (mbox *Mailbox) DeleteEmails(seqs []uint32) error {
+	imapClient, err := mbox.newClient()
 	if err != nil {
-		return &imap.Command{}, fmt.Errorf("uid search failed: %s", err)
+		return err
 	}
-	return cmd, nil
-}
 
-const GenerateBufferSize = 100
+	defer imapClient.Logout()
 
-func (mbox *Mailbox) generateMail(search string, since *time.Time, markAsRead, delete bool) (chan Response, error) {
-	responses := make(chan Response, GenerateBufferSize)
-	client, err := mbox.newClient()
+	seqSet := new(imap.SeqSet)
+	seqSet.AddNum(seqs...)
+
+	item := imap.FormatFlagsOp(imap.AddFlags, true)
+	err = imapClient.Store(seqSet, item, imap.DeletedFlag, nil)
 	if err != nil {
-		close(responses)
-		return responses, fmt.Errorf("failed to create IMAP connection: %s", err)
+		return err
 	}
 
-	go func() {
-		defer func() {
-			client.Close(true)
-			client.Logout(30 * time.Second)
-			close(responses)
-		}()
+	return nil
+}
 
-		var cmd *imap.Command
-		// find all the UIDs
-		cmd, err = findEmails(client, search, since)
-		if err != nil {
-			responses <- Response{Err: err}
-			return
-		}
-		// gotta fetch 'em all
-		getEmails(client, cmd, markAsRead, delete, responses)
-	}()
+// GetUnread will find all unread emails in the folder and return them as a list.
+func (mbox *Mailbox) GetUnread(markAsRead, delete bool) ([]Email, error) {
 
-	return responses, nil
-}
+	imap.CharsetReader = charset.Reader
+	var emails []Email
 
-func getEmails(client Client, cmd *imap.Command, markAsRead, delete bool, responses chan Response) {
-	seq := &imap.SeqSet{}
-	msgCount := 0
-	for _, rsp := range cmd.Data {
-		for _, uid := range rsp.SearchResults() {
-			msgCount++
-			seq.AddNum(uid)
-		}
+	imapClient, err := mbox.newClient()
+	if err != nil {
+		return emails, fmt.Errorf("failed to create IMAP connection: %s", err)
 	}
 
-	if seq.Empty() {
-		return
-	}
+	defer imapClient.Logout()
 
-	fCmd, err := imap.Wait(client.UIDFetch(seq, "INTERNALDATE", "BODY[]", "UID", "RFC822.HEADER"))
+	// Search for unread emails
+	criteria := imap.NewSearchCriteria()
+	criteria.WithoutFlags = []string{"\\Seen"}
```

Comment:
I like to defer to package constants where possible. It looks like the package has a [`SeenFlag="\\Seen`](https://godoc.org/github.com/emersion/go-imap#pkg-constants) that we might be able to use here.

Here's what that might look like (please double-check me though 😄)

```suggestion
	criteria.WithoutFlags = []string{imap.SeenFlag}
```

Reply:
Done.